org.jacoco.core.test/src/org/jacoco/core/test/InstrumentingLoader.java

```diff
@@ -21,31 +20,53 @@
 import org.jacoco.core.runtime.RuntimeData;
 import org.jacoco.core.runtime.SystemPropertiesRuntime;
 
+/**
+ * Class loader which loads classes from another class loader and instruments
+ * them.
+ */
 public final class InstrumentingLoader extends ClassLoader {
 
-	private final RuntimeData data;
 	private final IRuntime runtime;
+	private final String scope;
+	private final ClassLoader delegate;
 
-	public InstrumentingLoader() throws Exception {
-		data = new RuntimeData();
-		runtime = new SystemPropertiesRuntime();
+	private final RuntimeData data;
+	private final Instrumenter instrumenter;
+
+	public InstrumentingLoader(IRuntime runtime, String scope,
```

Comment:
@marchof after removal of unused constructor this one will be used only from the another remaining